doc/fundamentals/distributed_computing.md

```diff
@@ -0,0 +1,115 @@
+
+2) How to define which workers are responsible for which data?
+
+## Pachyderm Workers
+
+Before we dive into the above questions, there are a few details you should understand about Pachyderm workers. 
+
+Pachyderm workers are spun up when you create the pipeline and are left running in the cluster waiting for new jobs (data) to be available for processing (committed). This saves having to recreate and schedule the worker pod for every new job.
+
+Every worker for a given pipeline is an identical pod running the Docker image you specified in the [pipeline spec](../reference/pipline_spec.html). Your analysis code does not need do anything special to run in a distributed fashion. Instead, Pachyderm will spread out the data that needs to be processed across the various workers and make that data available for your code. 
+
+## Controlling the Number of Workers (Parallelism)
+
+The number of workers that are used for a given pipeline is controlled by the `parallelism_spec` defined in the pipeline specification.
+
+```
+  "parallelism_spec": {
+    "strategy": "CONSTANT"|"COEFFICIENT"
+    "constant": int        // if strategy == CONSTANT
+    "coefficient": double  // if strategy == COEFFICIENT
+```
+Pachyderm has two parallelism strategies: `CONSTANT` and `COEFFICIENT`.
+
+If you use the `CONSTANT` strategy, Pachyderm will start the number of workers that you specify. To use this strategy, set the field strategy to `CONSTANT`, and set the field `constant` to an integer value (e.g. `"constant":10` to use 10 workers).
+
+If you use the `COEFFICIENT` strategy, Pachyderm will start a number of workers that is a multiple of your Kubernetes cluster’s size. To use this strategy, set the field `coefficient` to a double. For example, if your Kubernetes cluster has 10 nodes, and you set `"coefficient": 0.5`, Pachyderm will start five workers. If you set it to 2.0, Pachyderm will start 20 workers (two per Kubernetes node).
+
+**NOTE:** The parallelism_spec is optional and will default to `“coefficient": 1”`, which means that it'll spawn one worker per Kubernetes node for this pipeline. 
+
+## Sharding Data Across Workers (Glob Patterns)
+
+Defining how your data is spread out among workers is arguably the most important aspect of distributed computation and is the fundamental idea around concepts like MapReduce. 
+
+Instead of confining users to data sharding patterns such as Map (split everything as much as possible) and Reduce (all the data must be grouped together), Pachyderm uses [Glob Patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to offer incredible flexibility in defining your data distribution. 
+
+Glob patterns are defined for each input of a pipeline. 
+
+```
+"inputs": [
+    {
+      "repo": {
+        "name": "string"
+      },
+      "glob": "string"
+    }
+  ]
+```
+That means you could easily define two inputs, one with the data highly distributed and another where it's grouped together. 
+
+### Datums
```

Comment:
maybe add a brief statement above this section about why we are going to talk about datums etc.  Just to guide the reader.